src/main/java/com/rescuebites/api/users/services/implementations/UserServiceImpl.java

```diff
@@ -35,20 +35,20 @@ public class UserServiceImpl implements IUserService {
     @Override
     public void saveUser(RegisterRequest registerRequest) {
         verifyIfEmailAlreadyExists(registerRequest.email());
-        verifyIfPasswordsMatch(registerRequest);
+        verifyIfPasswordsMatch(registerRequest.password(), registerRequest.confirmPassword());
 
         User newUser = userMapper.toUser(registerRequest);
         newUser.setPassword(passwordEncoder.encode(registerRequest.password()));
         userRepository.save(newUser);
 
         //Generamos el token de confirmación y enviamos el email
         UUID confirmationToken = tokenService.saveUserToken(newUser).getTokenId();
-        String confirmAccountHtml = emailBuilder.buildConfirmAccount(newUser.getEmail(), confirmationToken);
+        String confirmAccountHtml = emailBuilder.buildConfirmAccount(newUser, confirmationToken);
         emailService.sendEmail(newUser.getEmail(), "Confirm your registration ✔", confirmAccountHtml);
     }
 
-    private void verifyIfPasswordsMatch(RegisterRequest registerRequest) {
-        if(!registerRequest.password().equals(registerRequest.confirmPassword())){
+    private void verifyIfPasswordsMatch(String password, String confirmPassword) {
+        if(!password.equals(confirmPassword)){
             throw new IllegalArgumentException("Las contraseñas no coinciden");
```

Comment:
Si bien este tipo de excepción devuelve un HTTP Status Code 400, no es del todo correcto utilizar este tipo de excepciones para validaciones de datos por reglas de negocio. Para esto es mejor utilizar un ResponseStatusException o custom exceptions


```suggestion
            throw new PasswordsDoesNotMatchException();
```

donde `PasswordsDoesNotMatchException` es una clase del tipo exception. Acá te dejo una guía al respecto

https://www.baeldung.com/java-new-custom-exception